akka-persistence-query/src/main/scala/akka/persistence/query/journal/leveldb/EventsByPersistenceIdPublisher.scala

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.query.journal.leveldb
+
+import scala.concurrent.duration._
+import akka.actor.ActorLogging
+import akka.actor.ActorRef
+import akka.actor.Props
+import akka.persistence.JournalProtocol._
+import akka.persistence.Persistence
+import akka.stream.actor.ActorPublisher
+import akka.stream.actor.ActorPublisherMessage.Cancel
+import akka.stream.actor.ActorPublisherMessage.Request
+import akka.persistence.journal.leveldb.LeveldbJournal
+import akka.persistence.query.EventEnvelope
+
+/**
+ * INTERNAL API
+ */
+private[akka] object EventsByPersistenceIdPublisher {
+  def props(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long, refreshInterval: Option[FiniteDuration],
+            maxBufSize: Int, writeJournalPluginId: String): Props =
+    Props(new EventsByPersistenceIdPublisher(persistenceId, fromSequenceNr, toSequenceNr, refreshInterval,
+      maxBufSize, writeJournalPluginId))
+
+  private case object Continue
+}
+
+class EventsByPersistenceIdPublisher(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long,
+                                     refreshInterval: Option[FiniteDuration],
+                                     maxBufSize: Int, writeJournalPluginId: String)
+  extends ActorPublisher[EventEnvelope] with DeliveryBuffer[EventEnvelope] with ActorLogging {
+  import EventsByPersistenceIdPublisher._
+
+  val journal: ActorRef = Persistence(context.system).journalFor(writeJournalPluginId)
```

Comment:
I'm not sure what you mean. In reference.conf there is a `write-plugin` property that is used for connecting the read with the write side. This must indeed be documented.

For shared, I think it just works. The ref will be the proxy.